tests/virt_autotest/switch_version_and_reload_needle.pm

```diff
@@ -0,0 +1,29 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: This test changes the VERSION to TARGET_DEVELOPING_VERSION and reload needles.
+#          This test should be loaded after host upgrade.
+#          It needs to be wrapped in test because
+#          set_var with reload_needles option needs to be run in test files rather than main.pm,
+# Maintainer: xlai@suse.com
+
+
+use strict;
+use warnings;
+use base "virt_autotest_base";
+use testapi;
+
+sub run {
+    #switch VERSION TO TARGET_DEVELOPING_VERSION
+    die 'Var TARGET_DEVELOPING_VERSION is not set when switching VERSION to it!' unless get_var('TARGET_DEVELOPING_VERSION');
```

Comment:
you can use get_required_var - that does that for you

Reply:
will change to it